### PR TITLE
Adding `neo4j-rust-ext` recipe

### PR DESCRIPTION
**Pull request by robsdedude**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).  
  (It is vendoring packages in as far as Rust links dependencies statically)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.  
  (only me, you may take me opening the PR as my consent :wink:)
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by robsdedude** [comment]
@conda-forge-admin, please ping conda-forge/help-rust

Ready for review.

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/neo4j-python-driver-rust-ext/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+{% set pypi_name = "neo4j-rust-ext" %}
+{% set name = "neo4j-python-driver-rust-ext" %}
+{% set version = "5.26.0.0" %}
+{% set python_min = "3.7" %}
```
[the remark this answers is not in the document]
Comment (robsdedude) [reply recipes/neo4j-python-driver-rust-ext/meta.yaml:4]: Thanks for the review. I'm curious to learn why this is? I'm aware that support for Python 3.8 and older has been dropped from conda-forge. Is this all? I figured it'd still be a good idea to explicitly specify the minimum Python version as it wouldn't break anything and, well, be explicit 😅

[the remark this answers is not in the document]
Comment (robsdedude) [reply recipes/neo4j-python-driver-rust-ext/meta.yaml]: :facepalm: I just realized I didn't even use the variable :roll_eyes:

> as it wouldn't break anything
Turned out it was more struggle than I thought to get this right. So I just removed the Python version constraint all together.

**Comment by robsdedude** [comment]
@xhochy applied all your suggestions. But even after reverting everything to where it was before these changes, the mac os build fails with some 404 error. I don't understand what's wrong. Can you provide guidance (also regarding the open question on your review comment)?

**Review by xhochy — APPROVED** [review]

**Pull request merged**